cli/src/commands/create/dataset.rs

```diff
@@ -1,15 +1,30 @@
 use failchain::ResultExt;
+use lazy_static::lazy_static;
 use log::info;
+use regex::Regex;
 use reinfer_client::{
     resources::comment::EntityKind, Client, DatasetFullName, NewDataset, SourceIdentifier,
 };
 use structopt::StructOpt;
 
 use crate::errors::{ErrorKind, Result};
 
+lazy_static! {
+    static ref RX_FULL_NAME: Regex =
+        Regex::new("^[A-Za-z0-9-_.]{1,1024}/[A-Za-z0-9-_]{1,256}").unwrap();
+}
```

Comment:
Optional: we're tending towards using `once_cell` instead of `lazy_static`: no need for macro magic and also `once_cell` is potentially on the way to the standard library.

Could use `once_cell::sync::Lazy` if you fancy trying it out. (Or indeed migrate the whole cli project to `once_cell` in a separate PR?)

Reply:
I think converting this here to use `once_cell` rather than `lazy_static` is a good idea, and I'll do it in this PR.

Should probably leave conversion of the rest of the project to `once_cell` to a different PR like you say!